lib/puppet/provider/puppet_certificate/ruby.rb

```diff
@@ -165,8 +165,15 @@ def clean
     https.key = OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
     https.verify_mode = OpenSSL::SSL::VERIFY_PEER
     https.ca_file = Puppet.settings[:localcacert]
-    resp = https.start { |cx| cx.request(req) }
-    warning "failed to clean certificate: #{resp.body}" if resp.code_type != Net::HTTPNoContent
+    begin
+      resp = https.start { |cx| cx.request(req) }
+      warning "failed to clean certificate: #{resp.body}" if resp.code_type != Net::HTTPNoContent
+    rescue OpenSSL::SSL::SSLError => e
+      raise unless e.message.include?('certificate unknown')
```

Comment:
could the error also occur when the agent doesn't trust the server/CA certificate because it changed, for whatever reasons? Would that be a possible solution?

Reply:
Perhaps??  But, if you don't trust the CA anymore, then you're going to have issues getting your catalog in the first place.

Reply:
In my case, the agent cert was valid and the agent had an up to date CA cert.  Just that for some reason, the agent cert was missing on the CA itself.